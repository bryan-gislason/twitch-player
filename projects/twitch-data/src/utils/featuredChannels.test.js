import { extractFeaturedChannels } from ".";

describe("extractFeaturedChannels", () => {
  const { data } = require("../../fixtures/featuredChannels.json");
  const similarData = {
    ...data,
    channels: data.channels.slice(),
  };
  const dissimilarData = {
    ...data,
    channels: data.channels.slice(1),
  };

  it("extracts the channel nodes from a response", () => {
    expect(extractFeaturedChannels(data)).toMatchSnapshot();
  });

  it("returns the same result for the exact same argument", () => {
    expect(extractFeaturedChannels(data)).toBe(extractFeaturedChannels(data));
  });

  it("returns the same result for similar argument", () => {
    expect(extractFeaturedChannels(data)).toBe(
      extractFeaturedChannels(similarData)
    );
  });

  it("returns a different result for dissimilar argument", () => {
    expect(extractFeaturedChannels(data)).not.toBe(
      extractFeaturedChannels(dissimilarData)
    );
  });
});
