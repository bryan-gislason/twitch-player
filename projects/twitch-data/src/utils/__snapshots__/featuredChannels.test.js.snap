// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractFeaturedChannels extracts the channel nodes from a response 1`] = `
Array [
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "Free Hugs Here! Front Page Hypu !prime  @Broskian_",
      },
      "displayName": "Broskian",
      "id": "30972653",
      "key": "30972653",
      "login": "broskian",
      "profileImageURL": "profiles/fd117af9-8198-4adb-a4c7-d845163d41be-profile_image-150x150.png",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Destiny%202-376x500.jpg",
      "followersCount": 3142857,
      "id": "497057",
      "key": "497057",
      "name": "Destiny 2",
      "viewersCount": 12808,
    },
    "id": "31879258576",
    "key": "31879258576",
    "previewImageURL": "channels/live_user_broskian-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "e90b5f6e-4c6e-4003-885b-4d0d5adeb580",
        "isLanguageTag": false,
        "key": "e90b5f6e-4c6e-4003-885b-4d0d5adeb580",
        "localizedName": "Family Friendly",
        "tagName": "Family Friendly",
      },
      Object {
        "__typename": "Tag",
        "id": "ac763b17-7bea-4632-9eb4-d106689ff409",
        "isLanguageTag": false,
        "key": "ac763b17-7bea-4632-9eb4-d106689ff409",
        "localizedName": "Playing with Viewers",
        "tagName": "Playing with Viewers",
      },
      Object {
        "__typename": "Tag",
        "id": "ff56eeeb-99ed-4a60-93fc-0b3f05d8661e",
        "isLanguageTag": false,
        "key": "ff56eeeb-99ed-4a60-93fc-0b3f05d8661e",
        "localizedName": "Multiplayer",
        "tagName": "Multiplayer",
      },
      Object {
        "__typename": "Tag",
        "id": "1eba3cfe-51cc-460a-8259-bc8bb987f904",
        "isLanguageTag": false,
        "key": "1eba3cfe-51cc-460a-8259-bc8bb987f904",
        "localizedName": "Competitive",
        "tagName": "Competitive",
      },
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 5374,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "6 DAYS OF ZELDA FOR CHARITY! - Zeldathon Response",
      },
      "displayName": "SuperMCGamer",
      "id": "21015760",
      "key": "21015760",
      "login": "supermcgamer",
      "profileImageURL": "profiles/afab8c8e-9e3c-47fe-a262-f376c3073eab-profile_image-150x150.png",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/./The%20Legend%20of%20Zelda:%20Ocarina%20of%20Time-376x500.jpg",
      "followersCount": 694628,
      "id": "11557",
      "key": "11557",
      "name": "The Legend of Zelda: Ocarina of Time",
      "viewersCount": 9044,
    },
    "id": "31870691600",
    "key": "31870691600",
    "previewImageURL": "channels/live_user_supermcgamer-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "bb5e7234-380e-48ad-a59c-03e51274e478",
        "isLanguageTag": false,
        "key": "bb5e7234-380e-48ad-a59c-03e51274e478",
        "localizedName": "Charity",
        "tagName": "Charity",
      },
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 7309,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "Pok√©mon starting 10AM every day! !pokemon",
      },
      "displayName": "TwitchPresents",
      "id": "149747285",
      "key": "149747285",
      "login": "twitchpresents",
      "profileImageURL": "profiles/dd542e0da09855b6-profile_image-150x150.jpeg",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Always%20On-376x500.jpg",
      "followersCount": 408719,
      "id": "499973",
      "key": "499973",
      "name": "Always On",
      "viewersCount": 13661,
    },
    "id": "31871633760",
    "key": "31871633760",
    "previewImageURL": "channels/live_user_twitchpresents-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "1706e401-a023-49fc-9b10-7bf73f59c940",
        "isLanguageTag": false,
        "key": "1706e401-a023-49fc-9b10-7bf73f59c940",
        "localizedName": "Marathon",
        "tagName": "Marathon",
      },
      Object {
        "__typename": "Tag",
        "id": "e3a6b378-232b-4ec2-9a82-86b72851e09a",
        "isLanguageTag": false,
        "key": "e3a6b378-232b-4ec2-9a82-86b72851e09a",
        "localizedName": "Animation",
        "tagName": "Animation",
      },
      Object {
        "__typename": "Tag",
        "id": "e90b5f6e-4c6e-4003-885b-4d0d5adeb580",
        "isLanguageTag": false,
        "key": "e90b5f6e-4c6e-4003-885b-4d0d5adeb580",
        "localizedName": "Family Friendly",
        "tagName": "Family Friendly",
      },
      Object {
        "__typename": "Tag",
        "id": "8a01ea18-df97-4046-9cff-a9a822bb96e5",
        "isLanguageTag": false,
        "key": "8a01ea18-df97-4046-9cff-a9a822bb96e5",
        "localizedName": "Closed Captions",
        "tagName": "Closed Captions",
      },
      Object {
        "__typename": "Tag",
        "id": "6606e54c-f92d-40f6-8257-74977889ccdd",
        "isLanguageTag": false,
        "key": "6606e54c-f92d-40f6-8257-74977889ccdd",
        "localizedName": "Anime",
        "tagName": "Anime",
      },
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 10176,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "We Gaming <3 ",
      },
      "displayName": "Ninja",
      "id": "19571641",
      "key": "19571641",
      "login": "ninja",
      "profileImageURL": "profiles/cef31105-8a6e-4211-a74b-2f0bbd9791fb-profile_image-150x150.png",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Fortnite-376x500.jpg",
      "followersCount": 35800783,
      "id": "33214",
      "key": "33214",
      "name": "Fortnite",
      "viewersCount": 240733,
    },
    "id": "31879434944",
    "key": "31879434944",
    "previewImageURL": "channels/live_user_ninja-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 25491,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "Continuing Year of Dragon! ",
      },
      "displayName": "Renae",
      "id": "62744891",
      "key": "62744891",
      "login": "renae",
      "profileImageURL": "profiles/cd34ce39-55dc-4c72-855c-26c6d49760fe-profile_image-150x150.png",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Spyro%20Reignited%20Trilogy-376x500.jpg",
      "followersCount": 47220,
      "id": "505380",
      "key": "505380",
      "name": "Spyro Reignited Trilogy",
      "viewersCount": 769,
    },
    "id": "31876018512",
    "key": "31876018512",
    "previewImageURL": "channels/live_user_renae-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 135,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "The Most Entertaining Follow @ProblyOvr9000",
      },
      "displayName": "ProblyOvr9000",
      "id": "76938196",
      "key": "76938196",
      "login": "problyovr9000",
      "profileImageURL": "profiles/331581fd-78bf-48af-897d-8474e97bcb4b-profile_image-150x150.png",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Battlefield%20V-376x500.jpg",
      "followersCount": 759020,
      "id": "504199",
      "key": "504199",
      "name": "Battlefield V",
      "viewersCount": 1373,
    },
    "id": "31879160288",
    "key": "31879160288",
    "previewImageURL": "channels/live_user_problyovr9000-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "8486f56b-8677-44f7-8004-000295391524",
        "isLanguageTag": false,
        "key": "8486f56b-8677-44f7-8004-000295391524",
        "localizedName": "PvP",
        "tagName": "PvP",
      },
      Object {
        "__typename": "Tag",
        "id": "1eba3cfe-51cc-460a-8259-bc8bb987f904",
        "isLanguageTag": false,
        "key": "1eba3cfe-51cc-460a-8259-bc8bb987f904",
        "localizedName": "Competitive",
        "tagName": "Competitive",
      },
      Object {
        "__typename": "Tag",
        "id": "ac763b17-7bea-4632-9eb4-d106689ff409",
        "isLanguageTag": false,
        "key": "ac763b17-7bea-4632-9eb4-d106689ff409",
        "localizedName": "Playing with Viewers",
        "tagName": "Playing with Viewers",
      },
      Object {
        "__typename": "Tag",
        "id": "dbe20395-0e44-4156-92de-4a77f6cdccf0",
        "isLanguageTag": false,
        "key": "dbe20395-0e44-4156-92de-4a77f6cdccf0",
        "localizedName": "Wargame",
        "tagName": "Wargame",
      },
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 73,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "HEWWO! Fun and Games GALORE! *What is your favorite Board Game?*",
      },
      "displayName": "PatrckStatic",
      "id": "40170801",
      "key": "40170801",
      "login": "patrckstatic",
      "profileImageURL": "profiles/9118f8c4-baa0-41c1-a0fd-2a37c2083ee6-profile_image-150x150.png",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Dead%20Space%202-376x500.jpg",
      "followersCount": 49559,
      "id": "23017",
      "key": "23017",
      "name": "Dead Space 2",
      "viewersCount": 223,
    },
    "id": "31878475872",
    "key": "31878475872",
    "previewImageURL": "channels/live_user_patrckstatic-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "8ba227ca-073c-46a7-b3cc-193e52c5ab4d",
        "isLanguageTag": false,
        "key": "8ba227ca-073c-46a7-b3cc-193e52c5ab4d",
        "localizedName": "Let's Play",
        "tagName": "Let's Play",
      },
      Object {
        "__typename": "Tag",
        "id": "cc8d5abb-39c9-4942-a1ee-e1558512119e",
        "isLanguageTag": false,
        "key": "cc8d5abb-39c9-4942-a1ee-e1558512119e",
        "localizedName": "Casual Playthrough",
        "tagName": "Casual Playthrough",
      },
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 213,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "Friday Eve !vid | @AceDolox1",
      },
      "displayName": "AceDolo",
      "id": "79505769",
      "key": "79505769",
      "login": "acedolo",
      "profileImageURL": "profiles/012fd354-2d3c-4518-8f41-a1bf34962a7b-profile_image-150x150.png",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/./Tom%20Clancy%27s%20Rainbow%20Six:%20Siege-376x500.jpg",
      "followersCount": 8852048,
      "id": "460630",
      "key": "460630",
      "name": "Tom Clancy's Rainbow Six: Siege",
      "viewersCount": 9968,
    },
    "id": "31878683072",
    "key": "31878683072",
    "previewImageURL": "channels/live_user_acedolo-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 100,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "back to the seas | official PVP | !420",
      },
      "displayName": "1cePrime",
      "id": "81795595",
      "key": "81795595",
      "login": "1ceprime",
      "profileImageURL": "profiles/49f9d82c-5067-4103-96c7-571d6ad22a57-profile_image-150x150.png",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Atlas-376x500.jpg",
      "followersCount": 77429,
      "id": "5945",
      "key": "5945",
      "name": "Atlas",
      "viewersCount": 56141,
    },
    "id": "31874641312",
    "key": "31874641312",
    "previewImageURL": "channels/live_user_1ceprime-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "ff56eeeb-99ed-4a60-93fc-0b3f05d8661e",
        "isLanguageTag": false,
        "key": "ff56eeeb-99ed-4a60-93fc-0b3f05d8661e",
        "localizedName": "Multiplayer",
        "tagName": "Multiplayer",
      },
      Object {
        "__typename": "Tag",
        "id": "8486f56b-8677-44f7-8004-000295391524",
        "isLanguageTag": false,
        "key": "8486f56b-8677-44f7-8004-000295391524",
        "localizedName": "PvP",
        "tagName": "PvP",
      },
      Object {
        "__typename": "Tag",
        "id": "2dd8ed44-9043-4457-bb8d-e6c9a35e0b4d",
        "isLanguageTag": false,
        "key": "2dd8ed44-9043-4457-bb8d-e6c9a35e0b4d",
        "localizedName": "Hair Styling",
        "tagName": "Hair Styling",
      },
      Object {
        "__typename": "Tag",
        "id": "a4fac2cc-7cd4-44a6-b620-178182389a5b",
        "isLanguageTag": false,
        "key": "a4fac2cc-7cd4-44a6-b620-178182389a5b",
        "localizedName": "Exploration",
        "tagName": "Exploration",
      },
      Object {
        "__typename": "Tag",
        "id": "ac763b17-7bea-4632-9eb4-d106689ff409",
        "isLanguageTag": false,
        "key": "ac763b17-7bea-4632-9eb4-d106689ff409",
        "localizedName": "Playing with Viewers",
        "tagName": "Playing with Viewers",
      },
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 229,
  },
  Object {
    "__typename": "Stream",
    "broadcaster": Object {
      "__typename": "User",
      "broadcastSettings": Object {
        "__typename": "BroadcastSettings",
        "title": "Dead Daylight Deademption 2 #charitybits  !Mentalhealth @Swag_Dracula",
      },
      "displayName": "Swag_Dracula",
      "id": "158636407",
      "key": "158636407",
      "login": "swag_dracula",
      "profileImageURL": "profiles/6a52e82a1f62a1f8-profile_image-150x150.png",
    },
    "category": Object {
      "__typename": "Game",
      "boxArtURL": "categories/Dead%20by%20Daylight-376x500.jpg",
      "followersCount": 4815438,
      "id": "491487",
      "key": "491487",
      "name": "Dead by Daylight",
      "viewersCount": 16126,
    },
    "id": "31879568144",
    "key": "31879568144",
    "previewImageURL": "channels/live_user_swag_dracula-800x450.jpg",
    "tags": Array [
      Object {
        "__typename": "Tag",
        "id": "1eba3cfe-51cc-460a-8259-bc8bb987f904",
        "isLanguageTag": false,
        "key": "1eba3cfe-51cc-460a-8259-bc8bb987f904",
        "localizedName": "Competitive",
        "tagName": "Competitive",
      },
      Object {
        "__typename": "Tag",
        "id": "63e83904-a70b-4709-b963-a37a105d9932",
        "isLanguageTag": false,
        "key": "63e83904-a70b-4709-b963-a37a105d9932",
        "localizedName": "Cooperative",
        "tagName": "Cooperative",
      },
      Object {
        "__typename": "Tag",
        "id": "df262ff6-74d1-415b-942c-d584d44426c4",
        "isLanguageTag": false,
        "key": "df262ff6-74d1-415b-942c-d584d44426c4",
        "localizedName": "Epic Level Characters",
        "tagName": "Epic Level Characters",
      },
      Object {
        "__typename": "Tag",
        "id": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "isLanguageTag": true,
        "key": "6ea6bca4-4712-4ab9-a906-e3336a9d8039",
        "localizedName": "English",
        "tagName": "auto___lang_en",
      },
    ],
    "viewersCount": 387,
  },
]
`;
